fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/TreeLike.java

```diff
@@ -83,8 +87,11 @@ default Iterable<? extends T> inPreOrder() {
      * @param descentIntoPredicate a predicate that determines of the traversal enters the subtree underneath the
      *        current node
      * @return an {@link Iterable} of nodes
+     * @deprecated this is deprecated for performance reasons, to apply a filter, get a {@link TreeLike#iterator()} which
+     *             returns an {@link Iterator} with pre-order traversal.
      */
     @Nonnull
+    @Deprecated
```

Comment:
Please have this use your much better iterator.

Reply:
Method is replaced with a normal `Stream#filter` as now `TreeLike` can return a `Stream` via `TreeLike#preOrderStream()` method.